spectator/percentile_buckets.cc

```diff
@@ -0,0 +1,86 @@
+#include "percentile_buckets.h"
+#include <limits>
+
+namespace spectator {
+
+#include "percentile_bucket_values.inc"
+
+int64_t GetPercBucketValue(size_t i) { return kBucketValues.at(i); }
+
+size_t PercentileBucketIndexOf(int64_t v) {
```

Comment:
Since C++ has unsigned types, would it be better to use `uint64_t` so the type system does more sanity checking for the user? The larger values could just be folded into the last bucket.

Reply:
I've been trying to follow roughly the Google C++ coding standards, and they recommend against using unsigned types unless you want to manipulate bytes or rely on overflow behavior: https://google.github.io/styleguide/cppguide.html#Integer_Types - In particular, they say:

> Do not use an unsigned type merely to assert that a variable is non-negative